debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReader.java

```diff
@@ -638,7 +640,48 @@ protected void handleUpdateTableMetadata(Event event) {
         if (recordMakers.assign(tableNumber, tableId)) {
             logger.debug("Received update table metadata event: {}", event);
         } else {
-            logger.debug("Skipping update table metadata event: {}", event);
+            if (context.dbSchema().isTableMonitored(tableId)) {
+                EventHeaderV4 eventHeader = event.getHeader();
+
+                if (inconsistentSchemaHandlingMode == EventProcessingFailureHandlingMode.FAIL) {
+                    logger.error(
+                            "Schema inconsistency detected while processing binlog event at offset {}.{}" +
```

Comment:
"Schema inconsistency" sounds a bit vague. How about:

"Encountered change event for table <xyz> whose schema isn't known to this connector. One possible cause is an incomplete database history topic. Take a new snapshot in this case".

Also the event itself isn't logged anymore, I think we should keep doing that.